Sources/MapboxCoreNavigation/AlternativeRoute.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import MapboxDirections
+import MapboxNavigationNative
+import Turf
+
+/**
+ Representation of an alternative route with relation to the original.
+ 
+ This object contains main and alternative routes which are sharing same origin and destination points, but differ at some point.
+ */
+public class AlternativeRoute: Identifiable {
+    /// Alternative route identificator type
+    public typealias ID = UInt32
+    /// Breif stats of a route for travelling
+    public typealias Stats = (distance: LocationDistance, duration: TimeInterval)
+
+    /// Alternative route identificator.
+    ///
+    /// It is unique withing the same navigation session.
+    public let id: ID
+    /// Original (main) route data
+    public let indexedRouteResponse: IndexedRouteResponse
+    /// Intersection on the main route, where alternative route branches.
+    public let mainRouteIntersection: Intersection
+    /// Intersection on the alternative route, where it splits from the main route.
+    public let alternativeRouteIntersection: Intersection
+    /// Alternative route statistics, counting from the split point.
+    public let statsFromFork: Stats
+    /// Alternative route statistics, counting from it's origin.
+    public let statsFromOrigin: Stats
+
+    init?(mainRoute: Route, alternativeRoute nativeRouteAlternative: RouteAlternative) {
+        self.id = nativeRouteAlternative.id
+        guard let decoded = RerouteController.decode(routeRequest: nativeRouteAlternative.route.getRequestUri(),
+                                                     routeResponse: nativeRouteAlternative.route.getResponseJson()) else {
+            return nil
+        }
+
+        self.indexedRouteResponse = .init(routeResponse: decoded.routeResponse,
+                                          routeIndex: Int(nativeRouteAlternative.route.getRouteIndex()))
+
+        var legIndex = Int(nativeRouteAlternative.mainRouteFork.legIndex)
+        var segmentIndex = Int(nativeRouteAlternative.mainRouteFork.segmentIndex)
+
+        guard let mainIntersection = mainRoute.findIntersection(on: legIndex, by: segmentIndex) else {
+            return nil
+        }
+        self.mainRouteIntersection = mainIntersection
+
+        legIndex = Int(nativeRouteAlternative.alternativeRouteFork.legIndex)
+        segmentIndex = Int(nativeRouteAlternative.alternativeRouteFork.segmentIndex)
+
+        guard let alternativeIntersection = indexedRouteResponse.currentRoute?.findIntersection(on: legIndex, by: segmentIndex) else {
+            return nil
+        }
+        self.alternativeRouteIntersection = alternativeIntersection
+
+        self.statsFromFork = (nativeRouteAlternative.infoFromFork.distance,
+                              nativeRouteAlternative.infoFromFork.duration)
+        self.statsFromOrigin = (nativeRouteAlternative.infoFromStart.distance,
+                                nativeRouteAlternative.infoFromStart.duration)
+    }
+}
```

Comment:
I cannot imagine a case when this could be useful. Instead, adding possibility to create this manually could result in a false sense that it is the way to introduce user alternatives to the system

Reply:
User is able to add his alternatives by including it into the `RouteResponse` he is setting. `AlternativeRoute` is the route which was detected by our SDK, so it is not meant to be created outside

Reply:
The case might be that on the client-side there is a class with alternative routes related business logic and in unit tests they might want to create different `AlternativeRoute` objects.